open-api/rest-catalog-open-api.yaml

```diff
@@ -2513,6 +2536,7 @@ components:
         - $ref: '#/components/schemas/RemovePropertiesUpdate'
         - $ref: '#/components/schemas/SetStatisticsUpdate'
         - $ref: '#/components/schemas/RemoveStatisticsUpdate'
+        - $ref: '#/components/schemas/AppendDataFilesUpdate'
```

Comment:
I don't think that this should be part of the table update API and should be a separate endpoint instead.

The table API commits to the table every time it is called, but I think that the value of an API that can post new data files is that you don't necessarily need to commit every time. One of the main use cases for this feature is to be able to produce data files in parallel and commit them in groups, with the catalog handling the coordination to keep track of the files and eventually commit them to the table. I don't see how the update proposed here would be useful for that purpose.

Reply:
> The table API commits to the table every time it is called, but I think that the value of an API that can post new data files is that you don't necessarily need to commit every time. 

+1000

I was planning to bring this up as an extension after the initial change is merged, because to me this use case can be supported by:
1. support asynchronous update, where the UpdateTable returns a 202 accepted with a handle to check for update progress, and another API to check for progress completion status.
2. support batch commit, where a flag is added to say the specific UpdateTable request can be batch-committed instead of being immediately committed. As of today the REST specification does not really enforce a 1:1 relationship between the UpdateTable call vs the transaction in the table, that's why I went with this line of thought.

If we go with a new API, do you have any initial thoughts on what it could look like?

Reply:
I think it would be an `append` endpoint under a table that you can POST files to. Maybe something like this?

```
POST /v1/{prefix}/namespaces/{namespace}/tables/{table}/append
{
  "accept-delay-ms": 300000,
  "data-files": [...]
}
```
```
202
Location: /v1/{prefix}/namespaces/{namespace}/tables/{table}/status/{id}
```